include/SDL3/SDL_gpu.h

```diff
@@ -0,0 +1,2251 @@
+ * The swapchain texture should only be referenced by the command buffer used to acquire it.
+ * May return NULL under certain conditions. This is not necessarily an error.
+ * This texture is managed by the implementation and must not be freed by the user.
+ * You MUST NOT call this function from any thread other than the one that created the window.
+ *
+ * \param commandBuffer a command buffer
+ * \param window a window that has been claimed
+ * \param pWidth a pointer filled in with the swapchain width
+ * \param pHeight a pointer filled in with the swapchain height
+ * \returns a swapchain texture
+ *
+ * \since This function is available since SDL 3.x.x
+ *
+ * \sa SDL_GpuClaimWindow
+ * \sa SDL_GpuSubmit
+ * \sa SDL_GpuSubmitAndAcquireFence
+ */
+extern SDL_DECLSPEC SDL_GpuTexture *SDLCALL SDL_GpuAcquireSwapchainTexture(
+    SDL_GpuCommandBuffer *commandBuffer,
+    SDL_Window *window,
+    Uint32 *pWidth,
+    Uint32 *pHeight);
+
+/**
+ * Submits a command buffer so its commands can be processed on the GPU.
+ * It is invalid to use the command buffer after this is called.
+ *
+ * \param commandBuffer a command buffer
+ *
+ * \since This function is available since SDL 3.x.x
+ *
+ * \sa SDL_GpuAcquireCommandBuffer
+ * \sa SDL_GpuAcquireSwapchainTexture
+ * \sa SDL_GpuSubmitAndAcquireFence
+ */
+extern SDL_DECLSPEC void SDLCALL SDL_GpuSubmit(
+    SDL_GpuCommandBuffer *commandBuffer);
+
+/**
+ * Submits a command buffer so its commands can be processed on the GPU,
+ * and acquires a fence associated with the command buffer.
+ * You must release this fence when it is no longer needed or it will cause a leak.
+ * It is invalid to use the command buffer after this is called.
+ *
+ * \param commandBuffer a command buffer
+ * \returns a fence associated with the command buffer
+ *
+ * \since This function is available since SDL 3.x.x
+ *
+ * \sa SDL_AcquireCommandBuffer
+ * \sa SDL_GpuAcquireSwapchainTexture
+ * \sa SDL_GpuSubmit
+ * \sa SDL_GpuReleaseFence
+ */
+extern SDL_DECLSPEC SDL_GpuFence *SDLCALL SDL_GpuSubmitAndAcquireFence(
+    SDL_GpuCommandBuffer *commandBuffer);
+
+/**
+ * Blocks the thread until the GPU is completely idle.
+ *
+ * \param device a GPU context
+ *
+ * \since This function is available since SDL 3.x.x
+ *
+ * \sa SDL_GpuWaitForFences
+ */
+extern SDL_DECLSPEC void SDLCALL SDL_GpuWait(
+    SDL_GpuDevice *device);
+
+/**
+ * Blocks the thread until the given fences are signaled.
+ *
+ * \param device a GPU context
+ * \param waitAll if 0, wait for any fence to be signaled, if 1, wait for all fences to be signaled
+ * \param pFences an array of fences to wait on
+ * \param fenceCount the number of fences in the pFences array
+ *
+ * \since This function is available since SDL 3.x.x
+ *
+ * \sa SDL_GpuSubmitAndAcquireFence
+ * \sa SDL_GpuWait
+ */
+extern SDL_DECLSPEC void SDLCALL SDL_GpuWaitForFences(
+    SDL_GpuDevice *device,
+    SDL_bool waitAll,
+    SDL_GpuFence **pFences,
+    Uint32 fenceCount);
```

Comment:
Is there an option for a timeout parameter?

Reply:
If we do timeouts we should probably expose it in AcquireSwapchainTexture as well because that can also block. Honestly it's probably a good idea, the only platform where it sucks to implement that is D3D11 because it doesn't have real fences but we can just hack it with SDL_Timer.